### PR TITLE
A bunch of improvements for the import tool.

### PR DESCRIPTION
**Pull request by zlobober**
* Introduce a config file for optional fine-grained tuning and configuring (e.g. for ipv4/ipv6 address resolver configuration, or max_row_weight);
* Set data_weight_per_job = 1 for download map operation, as it almost never hurts;
* Consider only objects containing ".parquet" in their names when working with S3 to filter out unrelated stuff like "COMMIT_SUCCEEDED.txt";
* Attach libiconv.so if it is present in CWD (before that, import tool did not work at all in OSS build);
* config -> subset parameter renaming for HuggingFace importer (according to API specification at https://huggingface.co/docs/hub/api#get-apidatasetsrepoidparquet);
* Cosmetics and logging improvements.

### CODE REVIEW

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+#include "config.h"
 
 #include <yt/yt/library/arrow_parquet_adapter/arrow.h>
 
 #include <yt/yt/library/s3/client.h>
 
 #include <yt/yt/library/huggingface_client/client.h>
 
+#include <yt/yt/library/program/helpers.h>
+#include <yt/yt/library/program/config.h>
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:12]: Wrong order)

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 2]
```diff
@@ -32,8 +35,23 @@
...
+#include <yt/yt/core/concurrency/thread_pool_poller.h>
+
+#include <yt/yt/core/misc/fs.h>
+
+#include <yt/yt/core/net/address.h>
+#include <yt/yt/core/net/config.h>
+
+#include <yt/yt/core/ytree/convert.h>
+
+#include <yt/yt/core/yson/string.h>
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:47]: Swap it with `#include <yt/yt/core/ytree/convert.h>`

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 3]
```diff
@@ -32,8 +35,23 @@
...
+
+#include <iostream>
+
 namespace NYT::NTools::NImporter {
 
+using namespace NYson;
+using namespace NYTree;
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:54]: Sort this lines too please

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 4]
```diff
@@ -572,8 +595,10 @@ void ImportParquetFilesFromSource(
     auto writer = ytClient->CreateTableWriter<TNode>(metaInformationTable.Name());
     int fileIndex = 0;
     for (const auto& fileName : fileIds) {
-        writer->AddRow(TNode()(FileIdColumnName, fileName)(FileIndexColumnName, fileIndex));
-        ++fileIndex;
+        if (fileName.Contains(".parquet")) {
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:598]: Let's allow to configure it?

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 5]
```diff
@@ -689,25 +750,28 @@ void ImportParquetFilesFromS3(
         proxy,
         fileKeys,
         resultTable,
-        TSourceConfig({ .S3Config = s3Config }));
+        TSourceConfig({ .S3Config = s3Config }),
+        config);
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:754]: std::move

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 6]
```diff
@@ -719,7 +783,8 @@ void ImportParquetFilesFromHuggingface(
             .HuggingfaceConfig = THuggingfaceConfig{
                 .UrlOverride = urlOverride
             }
-        });
+        },
+        config);
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:787]: std::move

File: yt/yt/tools/import_table/main.cpp [hunk 7]
```diff
@@ -26,13 +35,17 @@ struct TOpts
         Opts.AddLongOption("format", "Format of files")
             .DefaultValue("parquet")
             .StoreResult(&Format);
+        Opts.AddLongOption("config", "YSON file with configuration for fine-grained tuning")
+            .DefaultValue("")
```
Comment (savnadya) [inline yt/yt/tools/import_table/main.cpp:39]: Let's make it optional? And store as `std::optional<TString>`?

**Review by savnadya — CHANGES_REQUESTED** [review]

File: yt/yt/tools/import_table/lib/import_table.cpp [hunk 8]
```diff
@@ -612,28 +637,49 @@ void ImportParquetFilesFromSource(
...
-    TNode secureVault;
-
-    if (sourceConfig.S3Config) {
-        secureVault["ACCESS_KEY_ID"] = GetEnv("ACCESS_KEY_ID");
-        secureVault["SECRET_ACCESS_KEY"] = GetEnv("SECRET_ACCESS_KEY");
-    } else if (sourceConfig.HuggingfaceConfig) {
-        secureVault["HUGGINGFACE_TOKEN"] = GetEnv("HUGGINGFACE_TOKEN");
+    // If libiconv.so exists next to us, assume that we are in OSS world
+    // and it has to be attached to the user jobs.
+    bool attachLibIconv = NFS::Exists("./libiconv.so");
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/import_table.cpp:642]: https://github.com/ytsaurus/ytsaurus/blob/main/util/system/execpath.h#L17

File: yt/yt/tools/import_table/lib/import_table.cpp (hunk 4 above)
Comment (zlobober) [reply yt/yt/tools/import_table/lib/import_table.cpp]: Introduce a regexp in the configuration.

File: yt/yt/tools/import_table/lib/config.h [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+public:
+    //! Singletons configuration to be used in the main importer process.
+    TSingletonsConfigPtr Singletons;
+    //! Singletons configuration to be used in the map-reduce jobs.
+    TSingletonsConfigPtr JobSingletons;
+
+    //! Maximum weight of a single row in the imported table.
+    i64 MaxRowWeight;
+
+    //! Regex for filtering out non-parquet files like successful commit markers in S3 and other unrelated stuff..
```
Comment (savnadya) [inline yt/yt/tools/import_table/lib/config.h:27]: too much dots)

**Comment by savnadya** [comment]
Shipped!

**Pull request closed without merging**